Added support for Visual Studio 2019

TheZoc opened the pull request:
This PR will allow the user to compile mdloader for Windows easily, using Visual Studio 2019.
(It probably works in Visual Studio 2017, but I don't have it installed right now).

I used the superb [getopt windows implementation by Ludvik Jerabek](https://www.codeproject.com/Articles/157001/Full-getopt-Port-for-Unicode-and-Multibyte-Microso), as it is LGPL and works as the perfect drop in replacement for linux implementation.

zvecr commented:
It should be noted that the project builds fine within MSYS2 (and the [QMK MSYS](https://github.com/qmk/qmk_distro_msys) bundle) so there is a small trade-off on maintainability, having to keep the 2 build systems in step.

I would be tempted to implement this more like [this](https://github.com/Massdrop/mdloader/compare/master...zvecr:feature/cmake?expand=1), swapping to cmake as to not introduce as many VS specific changes.

TheZoc commented:
I personally don't like cmake for a bunch of reasons, not to mention it's an overkill for a project this size.
On Windows, it will just be a hard to use wrapper around Microsoft tools.

By looking at the comparison you linked, that one just look like a rework of another PR again - you just added the same solution I added and replaced the visual studio project files with cmake.

As long as the getopt is there, it's easy to re-create the project in visual studio. It's a plain project with all the files added.

--

I'll refrain from commenting or contributing from now on: clearly our views are different on this project and I'm not the one being paid to maintain it. Good luck.
